homeassistant/components/edl21/sensor.py

```diff
@@ -39,7 +39,7 @@
 
 DOMAIN = "edl21"
 CONF_SERIAL_PORT = "serial_port"
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=3)
```

Comment:
Does this mean that the device will actually report a new measurement every third second or just that some measurements can be reported at that interval?

It's important that we don't allow the device to spam irrelevant measurements to the Home Assistant state machine. Most measurements are probably not interesting at a 3 second interval.

Reply:
Thank you for your reply! I completely agree with your point, that the device should not spam irrelevant measurements. 
Ideally, one could set the update frequency in the configuration of the edl21 integration as proposed in this thread https://community.home-assistant.io/t/edl21-add-update-interval/180909

However, I don't know, how one could achieve that. I would be very willing to help in developing that, with some hints, how that would be achievable.

Reply:
As long as the rework to config flow is not finished, you should consider to accept this PR as a polling intervall of 60s is way to long for energy monitoring.

Reply:
My question above hasn't been answered.

Reply:
The EDL21 devices push new messages every 2 to 5 seconds through their optical interface, depending on the model. Therefore polling is not the right way to access it anyway (afaik technically you can not poll the device), but this is a topic for a rework of the sensor integration. You should also not get duplicated messages when using a shorter polling intervall but a more detailed information about the energy consumption.

Reply:
So every 3 seconds we're potentially going to record a new state for power, voltage, phase angle etc, besides energy? That doesn't sound useful at all for the majority.

Based on the current information I'm not going to approve this PR. I'll ask for a second opinion though. You're welcome to clarify the situation more regarding what messages are coming in and at what frequency.

Reply:
It looks like the data is being pushed over the serial port to the integration.  What was the original reason for limiting accepting new data to every 60 seconds?

Reply:
It was added to limit excessive state reporting:
https://github.com/home-assistant/core/pull/27962#issuecomment-559828892

Reply:
From a user perspective I can not see the benefits of limiting this component to 60s, best would be to add a config option (which was declined in #66405) or do a rework to avoid polling and integrate config flow and let the user customize the time between two data record processings. In the Volkszähler project (which seems to be the reference for the 60s delay) you can change the polling rate to whatever you want.

Any chance that an experienced home assistant deloper would rework the component at least to the point that the user can choose how many data should be processed?

Reply:
> It was added to limit excessive state reporting: [#27962 (comment)](https://github.com/home-assistant/core/pull/27962#issuecomment-559828892)

From the original PR in 2316f7ace4f1c081211cf8a32c9b82329292e223 :
* [edl21] Add 1 minute throttle to the sensor

With https://github.com/home-assistant/core/pull/66405 we don't want additional options, but since there is no way to change this one since its an internal throttle I think it would be reasonable to add an option to disable the throttle but we would need to convert it to a config flow first. Unfortunately it looks like the effort to do that has gone stale: https://github.com/home-assistant/core/pull/66753

Given that I think it makes sense to leave it as-is for now since we don't want to fill the database with many state changes for most users.